src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -47,31 +41,36 @@ public async Task InvokeAsync(HttpContext context)
 
             // Handle requests as per recommendation at http://graphql.org/learn/serving-over-http/
             var httpRequest = context.Request;
-            var gqlRequest = new GraphQLRequest();
+            GraphQLRequest gqlRequest = null;
             GraphQLRequest[] gqlBatchRequest = null;
 
             var writer = context.RequestServices.GetRequiredService<IDocumentWriter>();
+            var cancellationToken = GetCancellationToken(context);
 
-            if (HttpMethods.IsGet(httpRequest.Method) || (HttpMethods.IsPost(httpRequest.Method) && httpRequest.Query.ContainsKey(GraphQLRequest.QueryKey)))
+            if (HttpMethods.IsGet(httpRequest.Method) ||
+                (HttpMethods.IsPost(httpRequest.Method) && httpRequest.Query.ContainsKey(GraphQLRequest.QueryKey)))
             {
-                ExtractGraphQLRequestFromQueryString(httpRequest.Query, gqlRequest);
+                gqlRequest = _deserializer.DeserializeFromQueryString(httpRequest.Query);
             }
             else if (HttpMethods.IsPost(httpRequest.Method))
             {
                 if (!MediaTypeHeaderValue.TryParse(httpRequest.ContentType, out var mediaTypeHeader))
                 {
-                    await WriteBadRequestResponseAsync(context, writer, $"Invalid 'Content-Type' header: value '{httpRequest.ContentType}' could not be parsed.").ConfigureAwait(false);
+                    await WriteBadRequestResponseAsync(context, writer, cancellationToken, $"Invalid 'Content-Type' header: value '{httpRequest.ContentType}' could not be parsed.").ConfigureAwait(false);
```

Comment:
We talked about that all `ConfigureAwait(false)` can be removed from ASP.NET Core projects.

Reply:
This project targets `netstandard2.0`, I assume to cover all previous versions of asp.net core 2.2 and down. Its only "aspnet" dependency is `Microsoft.AspNetCore.Hosting.Abstractions`, so I guess theoretically someone could use the middleware on a non-aspnetcore stack. 

Probbaly best we just leave all these? They were already all there before this PR.

Reply:
That makes sense, but not much. Look at the package name. In fact, targeting on netstandard2.0 is a necessary but not sufficient condition for the real possibility of using the package in not ASP.NET Core environment. `netstandard2.0` makes more sense in core GraphQL package. At the server project level, targeting netstandard2.0 is nothing more than a tribute to minimalism.
@joemcbride @pekkah What do you think?

Reply:
Let's worry about this another time. Can do it in a separate issue one day.

Reply:
Agree.